product-scenarios/4-single-sign-on/4.1-sso-for-web-app/4.1.1-sso-with-saml/src/test/java/org/wso2/identity/scenarios/sso/test/saml/SAMLSSOBySPConfigTestCase.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.identity.scenarios.sso.test.saml;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpResponse;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.apache.xml.security.signature.XMLSignature;
+import org.opensaml.common.xml.SAMLConstants;
+import org.opensaml.saml2.core.Assertion;
+import org.opensaml.saml2.core.AuthnRequest;
+import org.opensaml.saml2.core.Response;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.application.common.model.xsd.ServiceProvider;
+import org.wso2.carbon.identity.sso.saml.stub.types.SAMLSSOServiceProviderDTO;
+import org.wso2.identity.scenarios.commons.SAML2SSOTestBase;
+import org.wso2.identity.scenarios.commons.SAMLConfig;
+import org.wso2.identity.scenarios.commons.TestConfig;
+import org.wso2.identity.scenarios.commons.TestUserMode;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.wso2.identity.scenarios.commons.util.DataExtractUtil.getCookieFromResponse;
+import static org.wso2.identity.scenarios.commons.util.DataExtractUtil.getRedirectUrlFromResponse;
+import static org.wso2.identity.scenarios.commons.util.DataExtractUtil.isConsentRequested;
+import static org.wso2.identity.scenarios.commons.util.SSOUtil.sendPOSTConsentMessage;
+import static org.wso2.identity.scenarios.commons.util.SSOUtil.sendRedirectRequest;
+
```

Comment:
It's good to add a short description here,what's the test class is about.

Reply:
Fixed with https://github.com/wso2/product-is/pull/6898